frontera/contrib/backends/remote/codecs/msgpack.py

```diff
@@ -65,33 +68,34 @@ def __init__(self, request_model, response_model, *a, **kw):
         self._response_model = response_model
 
     def _response_from_object(self, obj):
-        return self._response_model(url=obj[0],
+        url = to_native_str(obj[0])
+        return self._response_model(url=url,
                                     status_code=obj[1],
                                     body=obj[3],
-                                    request=self._request_model(url=obj[0],
+                                    request=self._request_model(url=url,
                                                                 meta=obj[2]))
 
     def _request_from_object(self, obj):
-        return self._request_model(url=obj[0],
+        return self._request_model(url=to_native_str(obj[0]),
                                    headers=obj[1],
                                    cookies=obj[2],
                                    meta=obj[3])
 
     def decode(self, buffer):
         obj = unpackb(buffer)
-        if obj[0] == 'pc':
+        if obj[0] == b'pc':
```

Comment:
I think, usage of bytes type in encoders for message type will improve readability and redundancy. Because currently you rely on fact that msgpack after unpacking returns bytes (even if original string was native), that could change in the future, and it's non-obvious.